libyul/backends/evm/StackLayoutGenerator.h

```diff
@@ -37,6 +37,7 @@ struct StackLayout
 		/// The resulting stack layout after executing the block.
 		Stack exitLayout;
 	};
+	bool useFunctions = false;
```

Comment:
BTW, the only non-member function that needs this to know this flag is `cleanUnreachable()`. But does the CFG pruning actually have to be conditional on `useFunctions`? If it's ok to prune functions on EOF, it should be ok to prune them in general. It's probably redundant (they would likely be pruned at a later stage anyway) but should not break anything.

It would be simpler to have the CFG behave consistently between EOF and non-EOF.

Reply:
It's also code from legacy EOF branch. I changed it according to your comments. Because I enabled pruning unused functions also for legacy which make a lot of sense it probably will results with some tests failing for legacy bytecode. Also not sure if some of them assume that unused functions are present in the compilation results for legacy.

Reply:
It looks that it doesn't break anything for legacy. https://github.com/ethereum/solidity/pull/15592

Reply:
> I changed it according to your comments.

Well, partially. I see you moved the flag to `StackLayoutGenerator`, which is good, but please also move it from CFG to `ControlFlowGraphBuilder` and `OptimizedEVMCodeTransform`.